### PR TITLE
Update logger.mdx file, adding CloudWatch setup

### PR DESCRIPTION
**Pull request by ktmdan**
_Issue #, if available:_
None

_Description of changes:_
Add to the end of logger.mdx file
Setup for using AWSCloudWatchProvider including roles.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by josefaidt** [comment]
Hey @ktmdan :wave: apologies for the delay here!! Can you describe the use case for this solution in more detail?

**Comment by ktmdan** [comment]
This PR describes how to setup a website to log to CloudWatch using the Amplify logger.

**Comment by chrisbonifacio** [comment]
Hi @ktmdan, thanks for the quick response. We discussed this internally and agree that this example would be useful to have the docs. However, because of the use of the CLI override feature being prominently featured this seems a bit more advanced for a new customer. Perhaps the Logger utility page is not the best place for it. Instead, could you add this under Guides and link to it from the Logger page?

**Comment by ktmdan** [comment]
Updated PR moved to Guides and added a Link under logging.

**Comment by ktmdan** [comment]
Also cspell should ignore code IMO.

File: src/fragments/guides/utilities/logger-cloudwatch.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+## Using Cloudwatch
+
+You can setup logger to log to Cloudwatch:
+### Extend IAM permissions (without roles)
+
+
+```
+amplify custom project
```
Comment (josefaidt) [inline src/fragments/guides/utilities/logger-cloudwatch.mdx:8]:
```suggestion
amplify override project
```

File: src/fragments/guides/utilities/logger-cloudwatch.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+              "logs:CreateLogStream",
+              "logs:DescribeLogGroups",
+              "logs:DescribeLogStreams",
+              "logs:CreateLogGroup"
+            ],
+            "Resource": "arn:aws:logs:*:<ACCOUNT ID>:log-group:weblogs"
```
Comment (josefaidt) [inline src/fragments/guides/utilities/logger-cloudwatch.mdx:40]: can we create a reference to the account ID here to mitigate the need of hardcoding? perhaps `cdk.Stack.of(this).account`?

File: src/fragments/guides/utilities/logger-cloudwatch.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+### Extend Permissions (with roles)
+
+Run this command and select userPoolGroups
+```
+amplify auth override
+```
+
+Modify the file amplify/backend/auth/userPoolGroups/override.ts with this (be sure to replace your account id):
```
Comment (josefaidt) [inline src/fragments/guides/utilities/logger-cloudwatch.mdx:57]:
```suggestion
Modify the file `amplify/backend/auth/userPoolGroups/override.ts` with the following (be sure to replace your account id):
```

File: src/fragments/guides/utilities/logger-cloudwatch.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  ];
+}
+
+```
+
+### Extend Permissions (with roles)
+
+Run this command and select userPoolGroups
+```
+amplify auth override
```
Comment (josefaidt) [inline src/fragments/guides/utilities/logger-cloudwatch.mdx:54]:
```suggestion
amplify override auth
```

minor nit, either works, but `amplify <action> <category>` is primarily used elsewhere

**Review by josefaidt — CHANGES_REQUESTED** [review]
Hey @ktmdan apologies for the delay here! I left a few comments, and was also curious as to whether this can be expanded to be included in the guides for other languages/platforms as well? This topic appears framework/language-agnostic

File: src/fragments/lib/utilities/js/logger-cloudwatch.mdx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+## Cloudwatch Logging
+
+For a guide on how to set up Cloudwatch Logging, visit [Cloudwatch Logging](/guides/utilities/logging)
```
Comment (josefaidt) [inline src/fragments/lib/utilities/js/logger-cloudwatch.mdx:3]:
```suggestion
## CloudWatch Logging

For a guide on how to set up CloudWatch Logging, visit the [CloudWatch Logging guide](/guides/utilities/logging)
```

File: src/fragments/guides/utilities/logger-cloudwatch.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+import awsconfig from "../aws-exports";
+
+Amplify.configure({
+  ...awsconfig,
+  Logging: {
+    logGroupName: "weblogs",
+    logStreamName: "<CHANGE STREAM NAME>"
+  }
+});
+export const CloudLogger = new Logger("app");
```
Comment (josefaidt) [inline src/fragments/guides/utilities/logger-cloudwatch.mdx:112]: can we include the `Logger` and `Amplify` class imports in this example to enable easy copy/paste?

**Comment by ktmdan** [comment]
This request was so long ago I no longer have the time to work on this.

**Comment by abdallahshaban557** [comment]
Hi @ktmdan  - apologies for not getting to your PR earlier. We are working on improving the logger (including our documentation) for developers needing to send logs to Cloudwatch.

**Pull request closed without merging**